sklearn/feature_selection/tests/test_from_model.py

```diff
@@ -40,6 +41,130 @@ def test_input_estimator_unchanged():
     assert_true(transformer.estimator is est)
 
 
+def check_invalid_max_features(est, X, y):
+    max_features = X.shape[1]
+    for invalid_max_n_feature in [-1, max_features + 1, 'gobbledigook']:
+        transformer = SelectFromModel(estimator=est,
+                                      max_features=invalid_max_n_feature,
+                                      threshold=-np.inf)
+        assert_raises(ValueError, transformer.fit, X, y)
```

Comment:
it's generally better to check the right error is being raised, especially for something as generic as a `ValueError`. Use `assert_raises_regexp` or `assert_raise_message`.